src/main/java/com/joss/conductor/mobile/ConductorConfig.java

```diff
@@ -385,4 +363,56 @@ public boolean isAutoGrantPermissions() {
     public void setAutoGrantPermissions(boolean autoGrantPermissions) {
         this.autoGrantPermissions = autoGrantPermissions;
     }
+
+    public Platform getPlatformName() {
+        return platformName;
+    }
+
+    public String[] getCurrentSchemes() {
+        return currentSchemes;
+    }
+
+    public String getFullAppPath() {
+        if (appFile == null) {
+            return null;
+        }
+
+        Path path = Paths.get(appFile);
+        if (appFile.startsWith("sauce-storage:") || path.isAbsolute()) {
+            return appFile;
+        }
+        try {
+            URL url = new URL(appFile);
+            return appFile;
+        } catch(MalformedURLException e) {
+            // Ignore, this is expected to happen, parse as a regular path
+        }
+
+        return Paths.get(System.getProperty("user.dir"), path.normalize().toString()).normalize().toString();
+    }
+
+    public Map<String, Object> getCustomCapabilities() {
+        return customCapabilities;
+    }
+
+    public void setCustomCapabilities(Object keyCustomCapabilities) {
+        Map<String, Object> custom = new HashMap<>();
+        if (keyCustomCapabilities instanceof Map) {
+            for (String key : ((Map<String, Object>) keyCustomCapabilities).keySet()) {
+                Object value = ((Map) keyCustomCapabilities).get(key);
+                if (!(value instanceof Map)) {
+                    custom.put(key, value);
+                } else {
+                    throw new ClassCastException(String.format("%s cannot be of type Map", value));
+                }
+            }
+        } else {
+            throw new ClassCastException(String.format("%s is expected to be a String list of key/value pairs", CUSTOM_CAPABILITIES));
+        }
+        setCustomCapabilities(custom);
+    }
+
+    public void setCustomCapabilities(Map<String, Object> customCapabilities) {
```

Comment:
The method name is a misnomer. It's adding custom capabilities, not setting them. I would either make the method name "addCustomCapabilities" or change what it does to replace what's currently there.

Reply:
So the reason I left it like this was so that "overriding" would work correctly with custom capabilities, but I think we can rename it to `putCustomCapabilities` so that overriding is implied.